src/client.jl

```diff
@@ -140,6 +139,8 @@ function setup_connection!(conn::Connection)
             namespace=conn.namespace)
     end
     Sockets.nagle(socket, false)  # Disables nagles algorithm. Appropriate for interactive connections.
+    # transmit session id
+    serialize(socket, conn.session_id)
```

Comment:
This needs to go after `verify_header()` and you should increment `PROTOCOL_VERSION` if you're making an incompatible change to the RemoteREPL protocol (such as adding a session ID)

Reply:
Agreed. done that. `send_header` and `verify_header` now come first.

It's an incompatible change from protocol perspective, but all user code is still backwards compatible. If no `UUID` is passed a random will be generated. I increased the `PROTOCOL_VERSION`.